### DISCUSSION TITLE
Unable to install agent

### DISCUSSION THREAD

**Discussion by JoachimVeulemans** [amidaware/tacticalrmm#1157] [Q&A]
Agent v.2.0.4
Docker-compose install v.0.13.4

The error I am getting is Get "https://rmmapi.domain/api/v3/installer/": EOF. Installation of mesh central without tactical rmm is working.

Output of check_mesh:
![image](https://user-images.githubusercontent.com/32447944/171202113-588af784-c7dc-4669-9de9-3c12e4f37887.png)

Output installer:
![image](https://user-images.githubusercontent.com/32447944/171202148-6bea899a-96eb-44a0-a617-c1f529a4b8c6.png)

Output chrome on agent:
![image](https://user-images.githubusercontent.com/32447944/171202184-09047c40-1128-4d54-bb99-bda1f487eac8.png)

I have another agent in the same network where the installation succeeds. I can not find a difference between the 2 agents. Would it be possible to add more info why the request failed during the install? Or what could be the issue here?

Many thanks in advance!

**Comment by silversword411** [comment]
Is there some kind of AV causing problems?

Client in the same network working vs not working should have same DNS, but might want to check that too.

**Reply by JoachimVeulemans** [reply]
It is a Windows 7 install without internet. I have disabled the firewall for testing, no luck. I think windows defender is active but I never encountered a problem with it regarding rmm. I will try and disable it tomorrow.

I have adapted the hosts file so that rmm.domain.com, rmmapi.domain.com and rmmmesh.domain.com are pointing to an internal server. (The agent will always stay in this network). Ping to rmmapi.domain.com is working and you can also see the server responding in Chrome.

When I look into the access logs of NGINX on the server, I can see the request from Chrome. When executing the installation command, I get the EOF and do not see any request arriving at the server, leading me to believe it is a problem on the agent. I just can't pinpoint it.

**Reply by silversword411** [reply]
Last person with the 0 error was bad dns

You tested dns from agent to server on all 3 urls?

**Reply by JoachimVeulemans** [reply]
Output ping:

![image](https://user-images.githubusercontent.com/32447944/171339446-0d8795c8-293e-4440-a6d5-8bb034b1d393.png)

**Reply by silversword411** [reply]
What's your SSL cert? It's not self-signed right?

**Reply by silversword411** [reply]
...and what version of browser is that....it looks....old.

**Reply by JoachimVeulemans** [reply]
No, it is not a self-signed certificate, it is one generated by LetsEncrypt. Because the agent does not have an internet connection, it can not install the new signing certificate that LetsEncrypt is using. Therefore, I installed "ISRG Root X1" manually. Maybe the certificate check fails during the install?

Chrome browser: v.43.0.2357.65

**Reply by silversword411** [reply]
You said you had 2 machines in the network, one worked one didn't...so it's something machine specific.

**Reply by silversword411** [reply]
Chrome v.43, that's what 4 years old? does that even support tls1.2?

**Reply by JoachimVeulemans** [reply]
I notice, in chrome it takes a long time for the url to respond, maybe 5 seconds. Could there be a timeout while installing?

**Reply by JoachimVeulemans** [reply]
It does support it, the server requires at least tls1.2.

**Reply by silversword411** [reply]
Yeah V43 is 2015.

Eol in 2020...I think your SSL suite is too old

**Reply by JoachimVeulemans** [reply]
![image](https://user-images.githubusercontent.com/32447944/171368074-222686bc-f497-4c05-83f0-0182ced27d29.png)

**Reply by silversword411** [reply]
Wait you're a docker...and trying to use a different mesh install with TRMM?

Maybe you should do a -nomesh install

**Reply by silversword411** [reply]
You should get on discord for more troubleshooting...but sounds like you have a very non-standard #unsupported install. You probably need paid support.

**Reply by JoachimVeulemans** [reply]
Yes, I am using Docker. I'm using the integrated Mesh Central container provided by TRMM.

When I use the Mesh Central Install without TRMM, everything works and I can take control so there is connectivity between the agent and the server:
![image](https://user-images.githubusercontent.com/32447944/171369754-1375f0df-7d33-4ed7-a449-f3211fcc39ef.png)

Executing with -nomesh gives the same error:
![image](https://user-images.githubusercontent.com/32447944/171370664-f423b2f6-c383-4f72-9335-19afab181d9d.png)

For me Mesh Central is not the problem here.

**Reply by silversword411** [reply]
Since it works from one machine and not another it's definitely machine specific.

Eof is a download that is interrupted midstream...which is why I said av in the beginning.

**Reply by JoachimVeulemans** [reply]
The only antivirus installed is the default Windows, but it seems to be disabled:

![image](https://user-images.githubusercontent.com/32447944/171415264-f97ee597-5dba-4985-b723-be8233472b18.png)

**Reply by JoachimVeulemans** [reply]
@silversword411 

I have been debugging some more. It seems that the connection to the server is too slow, and therefore throws a timeout.

Would it be possible to set the time when a timeout (command line argument) should happen, so I can adapt it?
Or would it be possible to add a command line argument where the installer is locally present?

Thanks in advance.

**Reply by JoachimVeulemans** [reply]
I have adapted the timeout in the NGINX server config. It is working now. Thanks for the help.

**Reply by wh1te909** [reply]
there actually used to be an optional installer flag you could pass called `-localmesh` and give it the path to the local meshagent but I removed it when we reworked the mesh installer to dynamically send the exe instead of you having to manually download/upload it from mesh's site lol. I can just add it back, would be good to have I guess.

**Reply by wh1te909** [reply]
@JoachimVeulemans actually nevermind on my last statement, the optional flag still exists I never removed it: https://github.com/amidaware/rmmagent/blob/959a1bb8c8f061f9279884fcad1d96d12e64fda6/agent/install.go#L170
